### ISSUE TITLE
IndexOutOfRange errors

### ISSUE THREAD

**Issue by airadam** [DigDes/SoapCore#714]
We're using SoapCore 1.1.0.10 in a .NET Core 3.1 project, running containerised in Linux, and have been seeing some odd problems. We have multiple endpoints, all doing different things, throwing errors of the following form;

```
System.IndexOutOfRangeException: Index was outside the bounds of the array.
at System.Xml.XmlNamespaceManager.LookupPrefix(String uri)
at SoapCore.Namespaces.AddNamespaceIfNotAlreadyPresentAndGetPrefix(XmlNamespaceManager xmlNamespaceManager, String preferredPrefix, String uri)
at SoapCore.CustomMessage.OnWriteStartEnvelope(XmlDictionaryWriter writer)
at System.ServiceModel.Channels.Message.WriteMessagePreamble(XmlDictionaryWriter writer)
at System.ServiceModel.Channels.Message.OnWriteMessage(XmlDictionaryWriter writer)
at SoapCore.MessageEncoder.SoapMessageEncoder.WriteXmlCore(Message message, XmlWriter xmlWriter)
at SoapCore.MessageEncoder.SoapMessageEncoder.WriteMessageAsync(Message message, PipeWriter pipeWriter)
at SoapCore.SoapEndpointMiddleware`1.WriteErrorResponseMessage(Exception exception, Int32 statusCode, IServiceProvider serviceProvider, Message requestMessage, SoapMessageEncoder messageEncoder, HttpContext httpContext)
at SoapCore.SoapEndpointMiddleware`1.ProcessOperation(HttpContext httpContext, IServiceProvider serviceProvider)
at SoapCore.SoapEndpointMiddleware`1.Invoke(HttpContext httpContext, IServiceProvider serviceProvider)
at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
at NewRelic.Providers.Wrapper.AspNetCore.WrapPipelineMiddleware.Invoke(HttpContext context)
```

These errors don't appear straight away - in fact, our service runs happily for hours - but once they start appearing, it seems to continue to worsen with time.

If there's something we could be getting wrong, please let me know. It does appear to be a SoapCore bug though, and one we urgently need to find a solution for.

This looks to be identical to the issues that were looked at in https://github.com/DigDes/SoapCore/pull/693/files

**Comment by tibel** [comment]
The mentioned PR is part of 1.1.0.12

Have you tested with the new versions already?

**Comment by airadam** [comment]
Thanks for your response.
At the time of our implementation, 1.1.0.10 was the latest version available; we haven't tried the more recent versions as yet, but if the fix is now included then we'll try 1.1.0.13 and see if there's any improvement after running for a few days. It's a tricky one as we don't have a way to guarantee triggering the fault for test purposes!
